SpaceX/NetworkServiceImp.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  NetworkServiceImp.swift
+//  SpaceX
+//
+//  Created by Nicolae Lupu on 30.09.2021.
+//
+
+import Foundation
+import UIKit
+
+enum HttpMethod: String {
+    case get = "GET"
+    case put = "PUT"
+    case post = "POST"
+}
+
+protocol NewtorkService: AnyObject {
+    func performRequest(urlComponents: URLComponents?, body: Data?,
+                        httpHeaders: [String : String]?, httpMethod: HttpMethod,
+                        completion: @escaping(Result <Data, Error>) -> Void)
+}
+
+class NetworkServiceImp: NewtorkService {
+    
+    let defaultSession = URLSession(configuration: .default)
+    let decoder = JSONDecoder()
+
+    func performRequest(urlComponents: URLComponents?,
+                        body: Data?,
+                        httpHeaders: [String: String]?,
+                        httpMethod: HttpMethod,
+                        completion: @escaping(Result <Data, Error>) -> Void) {
+        
+        guard let urlComponents = urlComponents,
+              let url = urlComponents.url else {
+                  preconditionFailure()
+              }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpBody = body
+        urlRequest.allHTTPHeaderFields = httpHeaders
+        urlRequest.httpMethod = httpMethod.rawValue
+        
+        let urlSession = defaultSession.dataTask(with: urlRequest, completionHandler: { data, response, error in
+            if let data = data {
+                completion(.success(data))
+            } else {
+                completion(.failure(error!))
+            }
+        })
```

Comment:
lets finish with this PR first, add The Custom error that implements Error protocol. 
```suggestion
        let urlSession = defaultSession.dataTask(with: urlRequest, completionHandler: { data, response, error in
            if let data = data {
                completion(.success(data))
            } else check the response status and app the error to custom Error
        })
```

Reply:
lets keep it simple in this PR just do 
```suggestion
        let urlSession = defaultSession.dataTask(with: urlRequest, completionHandler: { data, response, error in
            if let data = data {
                completion(.success(data))
            } else if let error = error {
                completion(.failure(error))
            }
        })
```

Reply:
and in the next PR we will add error handling